php/ext/google/protobuf/message.c

```diff
@@ -1129,33 +1128,47 @@ PHP_METHOD(Timestamp, fromDateTime) {
     return;
```

Comment:
You probably want to make this work with DateTimeImmutable as well, as that's how people *should* use it in PHP. I would recommend that instead of ``php_date_get_date_ce`` you use zpp as a val* and check with ``instanceof`` function whether datetime matches the ``php_date_get_date_ce()`` or ``php_date_get_immutable_ce`` return values.

Unfortunately, ``date_ce_interface`` is not exported through ``php_date_get_interface_ce()`` as that function doesn't exist.

Reply:
Good catch. Thx

Reply:
You don't need this code. You need to make this extension dependent on the "date" extension in your module's struct, similar to what we do here: https://github.com/derickr/mongo-php-driver/blob/master/php_phongo.c#L2872

The date extension is always available as @remicollet writes in https://github.com/google/protobuf/issues/3929#issuecomment-351288485 (where he also points to the module dependency). And with the date extension, ``php_date_get_date_ce()`` too.

What I was pointing at is that you really want to check for DateTimeInterface, which does *not* currently exports its CE. Which makes it not possible right now.

Reply:
Sorry for misunderstanding.
In the latest change, I made two thing:
1) Add optional dependency to date, because not every protobuf user needs fromDatetime/toDatetime.
2) Check DatetimeInterface class entry in fromDatetime. Now I use interface name to find class entry, should it work?